packages/amazonq/src/inlineChat/provider/inlineChatProvider.ts

```diff
@@ -0,0 +1,209 @@
+
+        if (triggerEvent.tabID === 'no-available-tabs') {
+            return
+        }
+
+        if (triggerEvent.tabID === undefined) {
+            setTimeout(() => {
+                this.generateResponse(triggerPayload, triggerID).catch((e) => {
+                    getLogger().error('generateResponse failed: %s', (e as Error).message)
+                })
+            }, 20)
+            return
+        }
+
+        const tabID = triggerEvent.tabID
+
+        const credentialsState = await AuthUtil.instance.getChatAuthState()
+
+        if (
+            !(credentialsState.codewhispererChat === 'connected' && credentialsState.codewhispererCore === 'connected')
+        ) {
+            await this.sendAuthNeededExceptionMessage(credentialsState, tabID, triggerID)
+            return
+        }
+        triggerPayload.useRelevantDocuments = false
+
+        const request = triggerPayloadToChatRequest(triggerPayload)
+        const session = this.sessionStorage.getSession(tabID)
+        getLogger().info(
+            `request from tab: ${tabID} conversationID: ${session.sessionIdentifier} request: ${JSON.stringify(
+                request
+            )}`
+        )
+
+        let response: GenerateAssistantResponseCommandOutput | undefined = undefined
+        session.createNewTokenSource()
+        try {
+            response = await session.chat(request)
+            getLogger().info(
+                `response to tab: ${tabID} conversationID: ${session.sessionIdentifier} requestID: ${response.$metadata.requestId} metadata: ${JSON.stringify(
+                    response.$metadata
+                )}`
+            )
+        } catch (e: any) {
+            this.processException(e, tabID)
+            this.errorEmitter.fire()
+            return undefined
+        }
+
+        return response
+    }
+
+    private processException(e: any, tabID: string) {
+        let errorMessage = ''
+        let requestID = undefined
+        const defaultMessage = 'Failed to get response'
+        if (typeof e === 'string') {
+            errorMessage = e.toUpperCase()
+        } else if (e instanceof SyntaxError) {
+            // Workaround to handle case when LB returns web-page with error and our client doesn't return proper exception
+            errorMessage = AwsClientResponseError.tryExtractReasonFromSyntaxError(e) ?? defaultMessage
+        } else if (e instanceof CodeWhispererStreamingServiceException) {
+            errorMessage = e.message
+            requestID = e.$metadata.requestId
+        } else if (e instanceof Error) {
+            errorMessage = e.message
+        }
+
+        this.errorEmitter.fire()
+        void vscode.window.showErrorMessage(errorMessage)
+        getLogger().error(`error: ${errorMessage} tabID: ${tabID} requestID: ${requestID}`)
+        this.sessionStorage.deleteSession(tabID)
+    }
+
+    private async sendAuthNeededExceptionMessage(credentialState: FeatureAuthState, tabID: string, triggerID: string) {
```

Comment:
we have too many different ways of showing a "auth needed" message. please re-use one of the existing ones, and generalize it if needed.

Reply:
Generalized it